test/es-module/test-cjs-esm-warn.js

```diff
@@ -6,36 +6,59 @@ const { spawn } = require('child_process');
 const assert = require('assert');
 const path = require('path');
 
-const requiring = path.resolve(fixtures.path('/es-modules/cjs-esm.js'));
-const required = path.resolve(
-  fixtures.path('/es-modules/package-type-module/cjs.js')
-);
+const requiringCjsAsEsm = path.resolve(fixtures.path('/es-modules/cjs-esm.js'));
+const requiringEsm = path.resolve(fixtures.path('/es-modules/cjs-esm-esm.js'));
 const pjson = path.resolve(
   fixtures.path('/es-modules/package-type-module/package.json')
 );
 
-const basename = 'cjs.js';
+{
+  const required = path.resolve(
+    fixtures.path('/es-modules/package-type-module/cjs.js')
+  );
+  const basename = 'cjs.js';
+  const child = spawn(process.execPath, [requiringCjsAsEsm]);
+  let stderr = '';
+  child.stderr.setEncoding('utf8');
+  child.stderr.on('data', (data) => {
+    stderr += data;
+  });
+  child.on('close', common.mustCall((code, signal) => {
+    assert.strictEqual(code, 1);
+    assert.strictEqual(signal, null);
+
+    assert.ok(stderr.replace(/\r/g, '').includes(
+      `Error [ERR_REQUIRE_ESM]: require() of ES Module ${required} from ${
+        requiringCjsAsEsm} not supported.\n`));
+    assert.ok(stderr.includes(
+      `Instead rename ${basename} to end in .cjs, change the requiring ` +
+      'code to use dynamic import() which is available in all CommonJS ' +
+      `modules, or remove "type": "module" from ${pjson} to ` +
```

Comment:
nit: shouldn't we advice to use something explicit instead?
```suggestion
      `modules, or change "type": "module" to "type": "commonjs" from ${pjson} to ` +
```

Reply:
I'm open to it. We aren't currently recommending `"type": "commonjs"`. It can be a good suggestion though since if eg npm did this we could switch the default for all packages. I'm not sure here is the right place to have that discussion though :P

Reply:
We kinda are already:
https://github.com/nodejs/node/blob/b7136827f5d9f4f65bd0a60e75c4a9531cf4312a/doc/api/packages.md#L78-L82

Reply:
Ok, sure let's do this then.